preview_generator/preview/builder/pdf__poppler_utils.py

```diff
@@ -0,0 +1,155 @@
+                    PDFTOCAIRO_EXECUTABLE,
+                    "-png",
+                    "-singlefile",
+                    "-scale-to",
+                    str(size.max_dim()),
+                    "-f",
+                    str(page_id),
+                    "-l",
+                    str(page_id),
+                    file_path,
+                    # HACK - G.M - 2021-10-21 - For unclear reason, pdftocairo add a second .png
+                    # extension to the file created.
+                    tmp_png.name.rsplit(".png", 1)[0],
+                ],
+                stdout=DEVNULL,
+                stderr=STDOUT,
+            )
+            if build_png_result_code != 0:
+                raise IntermediateFileBuildingFailed(
+                    "Building PNG intermediate file using pdftocairo failed with status {}".format(
+                        build_png_result_code
+                    )
+                )
+            return ImagePreviewBuilderPillow().build_jpeg_preview(
+                tmp_png.name, preview_name, cache_path, page_id, extension, size, mimetype
+            )
+
+    def build_pdf_preview(
+        self,
+        file_path: str,
+        preview_name: str,
+        cache_path: str,
+        extension: str = ".pdf",
+        page_id: int = -1,
+        mimetype: str = "",
+    ) -> None:
+        """
+        generate the pdf large preview
+        """
+        # INFO - G.M - 2021-10-21 - Page id in pdftocairo begin at 1 instead of 0
+        page_id = page_id + 1
+        preview_path = "{path}{file_name}{extension}".format(
+            file_name=preview_name, path=cache_path, extension=extension
+        )
+        if page_id > 0:
+            # page specific preview
+            check_call(
+                [
+                    PDFTOCAIRO_EXECUTABLE,
+                    "-pdf",
+                    "-f",
+                    str(page_id),
+                    "-l",
+                    str(page_id),
+                    file_path,
+                    preview_path,
+                ],
+                stdout=DEVNULL,
+                stderr=STDOUT,
+            )
+        else:
+            # Full preview
+            check_call(
+                [PDFTOCAIRO_EXECUTABLE, "-pdf", file_path, preview_path],
+                stdout=DEVNULL,
+                stderr=STDOUT,
+            )
+
+    def get_page_number(
+        self,
+        file_path: str,
+        preview_name: str,
+        cache_path: str,
+        mimetype: typing.Optional[str] = None,
+    ) -> int:
+        # TODO - G.M - 2021-10-21 - Reintroduce caching here ?
+        lines = check_output(["pdfinfo", file_path], stderr=STDOUT, universal_newlines=True).split(
+            "\n"
+        )
+        for line in lines:
+            section, data = line.split(":", maxsplit=1)
+            if section == "Pages":
+                return int(data.strip())
+        # FIXME - G.M - 2021-10-21 - Better default case ?
+        return 0
```

Comment:
@grignards @raphj This code replace the old code that does use a cached file. I'm not really such we do really improve performance between running pdfinfo command line or opening a cached file that store the number of page provided a first time by pdfinfo.
Did you think i should restore the previous behavior here or not ?

Reply:
Calling `pdfinfo` is probably slower, but it probably won't make much of a difference in the classic preview-generator applications.